### ISSUE TITLE
NRL get_response() is not calculating overall instrument_sensitivity correctly

### ISSUE THREAD

**Issue by mikehagerty** [obspy/obspy#3099]
### Avoid duplicates

- [X] I searched existing issues

### Bug Summary

When nrl calls get_response() to form the requested response, it calls response.recalculate_overall_sensitivity(), which in turn calls the evalresp library, but for some reason, it is not calculating the gain of the stage 1 poles_zeros correctly - it is not using A0 to calculate the gain at the normalization_frequency, and hence gives the wrong instrument_sensitivity.value.

The fix is to call response.recalculate_overall_sensivitity() a second time (why is this different ??)

The code below shows the problem:

1. When I call nrl.get_response(datalogger_keys, sensor_keys), it calls a function very similar to the function get_response_from_nrl() in the snippet below.  It joins the sensor + datalogger stages from the NRL and then calls response.recalculate_overall_sensitivity() to calculate/set response.instrument_sentivitity.value at the poles_zero normalization_frequency (5 Hz in this case).  However, that value is not correct - it does not take into account the A0 normalization of the stage 1 poles_zeros.  e.g., it is the value you would get if you assumed that the poles_zeros stage was normalized to 1.0 by the normalization_factor A0, and you just multiplied all of the response_stage.stage_gains together.  In the current example, expanding the poles_zeros at the normalization_frequency and scaling this by A0 gives:            |G(f=5.0Hz)| = 2.2068, thus the correct sensitivity should be: 6.29145492e+08 x 2.2068 = 1.38842473e+09 at 5.0 Hz

2. If I call nrl.get_response() and then call response.recalculate_overall_sensitivity() a second time (remember it is being called inside nrl.get_response()), it returns the correct value.

3a./3b. Here I am showing the effect of nrl.get_response() - it does call response.recalculate_overall_sensitivity() and this does change the sensitivity.value, but for some reason, again, it is not taking into account the A0 normalization of the poles_zeros stage.

3c. Again, a second call to response.recalculate_overall_sensitivity() fixes the problem.

Output of code:
** 1. straight from nrl.get_response():
Instrument Sensitivity:
	Value: 629145492.3989668
	Frequency: 5.0
	Input units: M/S
	Input units description: Velocity in Meters per Second
	Output units: COUNTS
	Output units description: Digital Counts

** 2. nrl.get_response() + resp.recalculate_overall_sensitivity():
Instrument Sensitivity:
	Value: 1388424726.398958
	Frequency: 5.0
	Input units: M/S
	Input units description: Velocity in Meters per Second
	Output units: COUNTS
	Output units description: Digital Counts

** 3a. get_response_from_nrl() initial resp:
Instrument Sensitivity:
	Value: 419430.375
	Frequency: 1.0
	Input units: M/S
	Input units description: Velocity in Meters per Second
	Output units: COUNTS
	Output units description: Digital Counts

** 3b. get_response_from_nrl() after first recalculate:
Instrument Sensitivity:
	Value: 629145492.3989668
	Frequency: 5.0
	Input units: M/S
	Input units description: Velocity in Meters per Second
	Output units: COUNTS
	Output units description: Digital Counts

** 3c. get_response_from_nrl() after additional recalculate:
Instrument Sensitivity:
	Value: 1388424726.398958
	Frequency: 5.0
	Input units: M/S
	Input units description: Velocity in Meters per Second
	Output units: COUNTS
	Output units description: Digital Counts

### Code to Reproduce

```python
from obspy.clients.nrl import NRL
def main():

    nrl = NRL()
    sensor_keys = ['GEObit', 'S20']
    datalogger_keys = ['GEObit','SRi32S Normal Gain', '40 Vpp', 'Linear Phase', '200']
    resp = nrl.get_response(datalogger_keys, sensor_keys)
    print("** 1. straight from nrl.get_response():\n%s" % resp.instrument_sensitivity)
    resp.recalculate_overall_sensitivity()
    print("** 2. nrl.get_response() + resp.recalculate_overall_sensitivity():\n%s" % resp.instrument_sensitivity)
    dl_resp = get_response_from_nrl(nrl, datalogger_keys, sensor_keys)

def get_response_from_nrl(nrl, datalogger_keys, sensor_keys):
    """ should be nearly identical to nrl.get_response() code """

    dl_resp = nrl.get_datalogger_response(datalogger_keys)
    sensor_resp = nrl.get_sensor_response(sensor_keys)

    dl_resp.response_stages.pop(0)
    sensor_stage0 = sensor_resp.response_stages[0]
    dl_resp.response_stages.insert(0, sensor_stage0)
    dl_resp.instrument_sensitivity.input_units = sensor_stage0.input_units
    dl_resp.instrument_sensitivity.input_units_description = sensor_stage0.input_units_description

    print("** 3a. get_response_from_nrl() initial resp:\n%s" % dl_resp.instrument_sensitivity)
    try:
        dl_resp.recalculate_overall_sensitivity()   ## This is not doing the trick!
    except ValueError:
        msg = "Failed to recalculate overall sensitivity"
        print(msg)
    print("** 3b. get_response_from_nrl() after first recalculate:\n%s" % dl_resp.instrument_sensitivity)

    dl_resp.recalculate_overall_sensitivity()
    print("** 3c. get_response_from_nrl() after additional recalculate:\n%s" % dl_resp.instrument_sensitivity)

    return dl_resp

if __name__ == '__main__':
    main()
```

### ObsPy Version?

1.3.0

### Operating System?

mac osx 12.4

### Python Version?

3.10.5

### Installation Method?

conda

**Comment by megies** [comment]
There is some weird stuff going on, can't make sense of it right now, but we need to look into details. It seems to also affect what evalresp is giving us as frequency response

**Comment by megies** [comment]
The problem seems to be that there is a mix of high level function calls where frequency is explicitly specified, but there are also quite a few low level operations involved that look at the response object and the stages itself. So by setting the frequency on the objects only at the very end, it seems that internally parts of the code still use a different frequency which is still not updated with the target frequency on the actual objects.

**Comment by megies** [comment]
Instead of only setting the frequency on the actual objects at the very end, just setting it right away at the start should fix the problem as done in #3141.

**Comment by megies** [comment]
@mikehagerty this should be fixed now for the upcoming bugfix release
